core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -527,12 +527,14 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       }
     }
 
-    // New consumer always hasNext
     override def hasData = true
 
     override def receive() : BaseConsumerRecord = {
-      while (recordIter == null || !recordIter.hasNext)
+      if (recordIter == null || !recordIter.hasNext) {
         recordIter = consumer.poll(1000).iterator
+        if (!recordIter.hasNext)
+          throw new ConsumerTimeoutException
```

Comment:
Not sure how worthwhile it would be, but we could use a static exception instance since we just throw it away when it gets caught anyway.